### PR TITLE
Feature/e UI 3848 cookie service 2 aa

### PR DESCRIPTION
**Pull request by adnanakgun**
Resolves https://tools.hmcts.net/jira/browse/EUI-3848

Created cookie service

### CODE REVIEW

**Review by ronaldmansveld — APPROVED** [review]

**Review by phillip-whitaker — APPROVED** [review]
A bit upset there were no fun cookie jokes in the tests, but will approve anyway.

File: projects/exui-common-lib/src/lib/services/cookie/cookie.service.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  private readonly document?: Document;
+
+  constructor(@Inject(DOCUMENT) doc?: any) {
+    this.document = doc as Document;
+  }
+
+  public setCookie(key: string, value: string, expiryDate?: string): void {
+    const expiry = expiryDate ? ` expires=${expiryDate}` : '';
+    const cookie = `${key}=${value};${expiry}`;
+    this.document.cookie = cookie;
```
[the remark this answers is not in the document]
Comment (paul-graham) [reply projects/exui-common-lib/src/lib/services/cookie/cookie.service.ts:15]: Fair play. I bet it won't work with your current test setup, though. 😉

**Review by paul-graham — APPROVED** [review]
Question about `document.cookie = cookie` has been addressed. I'm guessing it's a setter on the `document` that concatenates rather than replaces.

File: projects/exui-common-lib/src/lib/services/cookie/cookie.service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (adnanakgun) [reply projects/exui-common-lib/src/lib/services/cookie/cookie.service.ts:15]: Ah yes, probably not. Do you have a better suggestion to mock `document`?

File: projects/exui-common-lib/src/lib/services/cookie/cookie.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { CookieService } from './cookie.service';
+
+describe('CookieService', () => {
+  const mockDocument: any = {
+    cookie: ''
+  };
```
Comment (paul-graham) [inline projects/exui-common-lib/src/lib/services/cookie/cookie.service.spec.ts:6]: In response to the question about another way to mock the behaviour of the document on the browser:

```
class MockDocument {
  private _cookie: string;
  constructor() {
    this._cookie = '';
  }
  public set cookie(val: string) {
    this._cookie += `; ${val}`;
  }
  public get cookie(): string {
    return this._cookie;
  }
}
```

And then:
const mockDocument: MockDocument = new MockDocument();

File: projects/exui-common-lib/src/lib/services/cookie/cookie.service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (paul-graham) [reply projects/exui-common-lib/src/lib/services/cookie/cookie.service.ts:15]: https://github.com/hmcts/rpx-xui-common-lib/pull/133/files#r628340623

File: projects/exui-common-lib/src/lib/services/cookie/cookie.service.spec.ts (hunk 2 above)
Comment (paul-graham) [reply projects/exui-common-lib/src/lib/services/cookie/cookie.service.spec.ts:6]: Yeah, except it won't replace existing keys so it wouldn't work exactly as you'd expect. You'd probably want to use a dictionary instead... but it's just a mock!

**Comment by adnanakgun** [comment]
This all now in https://github.com/hmcts/rpx-xui-common-lib/pull/154 so am closing

**Pull request closed without merging**